lib/langchain/tool/google_search.rb

```diff
@@ -56,13 +56,78 @@ def self.execute_search(input:)
     def execute(input:)
       Langchain.logger.info("Executing \"#{input}\"", for: self.class)
 
-      hash_results = execute_search(input: input)
+      results = execute_search(input: input)
 
-      # TODO: Glance at all of the fields that langchain Python looks through: https://github.com/hwchase17/langchain/blob/v0.0.166/langchain/utilities/serpapi.py#L128-L156
-      # We may need to do the same thing here.
-      hash_results.dig(:answer_box, :answer) ||
-        hash_results.dig(:answer_box, :snippet) ||
-        hash_results.dig(:organic_results, 0, :snippet)
+      answer_box = results[:answer_box_list] ? results[:answer_box_list].first : results[:answer_box]
+      if answer_box
+        return answer_box[:result] ||
+            answer_box[:answer] ||
+            answer_box[:snippet] ||
+            answer_box[:snippet_highlighted_words] ||
+            answer_box.reject { |_k, v| v.is_a?(Hash) || v.is_a?(Array) || v.start_with?("http") }
+      elsif (events_results = results[:events_results])
+        return events_results.take(10)
+      elsif (sports_results = results[:sports_results])
+        return sports_results
+      elsif (top_stories = results[:top_stories])
+        return top_stories
+      elsif (news_results = results[:news_results])
+        return news_results
+      elsif (jobs_results = results.dig(:jobs_results, :jobs))
+        return jobs_results
+      elsif (shopping_results = results[:shopping_results]) && shopping_results.first.key?(:title)
+        return shopping_results.take(3)
+      elsif (questions_and_answers = results[:questions_and_answers])
+        return questions_and_answers
+      elsif (popular_destinations = results.dig(:popular_destinations, :destinations))
+        return popular_destinations
+      elsif (top_sights = results.dig(:top_sights, :sights))
+        return top_sights
+      elsif (images_results = results[:images_results]) && images_results.first.key?(:thumbnail)
+        return images_results.map { |h| h[:thumbnail] }.take(10)
+      end
+
+      snippets = []
+      if (knowledge_graph = results[:knowledge_graph])
+        snippets << knowledge_graph[:description] if knowledge_graph[:description]
+
+        title = knowledge_graph[:title] || ""
+        knowledge_graph.each do |k, v|
+          if v.is_a?(String) &&
+              k != :title &&
+              k != :description &&
+              !k.to_s.end_with?("_stick") &&
+              !k.to_s.end_with?("_link") &&
+              !k.to_s.start_with?("http")
+            snippets << "#{title} #{k}: #{v}"
+          end
+        end
+      end
+
+      if (first_organic_result = results.dig(:organic_results, 0))
+        if (snippet = first_organic_result[:snippet])
+          snippets << snippet
```

Comment:
Thank you, @tanys123! May we append snippets from all organic results?

```suggestion
      if (organic_results = results.dig(:organic_results))
        organic_results_snippets = organic_results.map { _1[:snippet] }.compact
        if (organic_results_snippets.size > 0)
          snippets.concat(organic_results_snippets)
```

(_I didn't test the suggested code above._)